hypothesis-python/tests/crosshair/test_crosshair.py

```diff
@@ -54,3 +59,46 @@ def f(n):
 
     with pytest.raises(BaseException, match="marker"):
         f()
+
+
+@pytest.mark.parametrize(
+    "strategy, use_observability, expected_choices",
+    [
+        (st.integers(), False, 1),
+        # we add an additional path constraint to ints in to_jsonable.
+        (st.integers(), True, 2),
+        (st.text(), False, 1),
+        (st.text(), True, 1),
+        (st.booleans(), False, 2),
+        (st.booleans(), True, 2),
+        (st.floats(), False, 6),
+        (st.floats(), True, 6),
+        (st.binary(), False, 1),
+        (st.binary(), True, 1),
```

Comment:
Rather than listing out the exact expected-number, can we check inside the test that the number of choices is invariant to no/observability and normal/verbose verbosity?  (obviously with a special case for `integers()`)

I'd rather not update the test just because e.g. Crosshair changed their float implementation.

Reply:
Then this test wouldn't catch bugs where we always add a path constraint to a drawn symbolic (under both normal and observability)! But I agree we shouldn't be changing this test for crosshair internals. I'll see if I can use crosshair to dynamically determine the "standard" number of choices for each type, which should be more stable than hardcoding.

Reply:
Feels to me like we should have a separate test for "expected number of constraints when Hypothesis didn't touch it", which could maybe work by using Crosshair directly to do an analogous thing?

We _could_ conflate these together, but I actually want different nodeids for "you broke it by making observability/verbosity touch stuff" vs "you're unconditionally adding constraints" 🤔

Reply:
I split this into two tests 👍. I actually found that we add a constraint to symbolic booleans inbetween when the provider returns it and we pass it to the test function, because of an internal `is` check on symbolic mappings: https://github.com/pschanely/CrossHair/blob/297f080873a06ae1113260f085a4c0d1eb181d0a/crosshair/simplestructs.py#L221. Which I'm writing off as "nothing we can do about it" (hypothesis-side, anyway) for the moment.